File: CloudFlow/FrontEnd-App/src/app/shared/model/registru-intrari-iesiri/registru-intrari-view.model.ts
import { JsonObject, JsonProperty } from "json2typescript";
import { JsonDateConverter } from "../../json-mapper";
import { DateUtils } from "../../utils";
import { RaspunsuriBanciCuPropuneriEnum, MonthEnum } from "../../enums";
import { DateConstants } from "../../constants";

@JsonObject
export class RegistruIntrariViewModel {

	@JsonProperty("id", Number)
	public id: number = null;

	@JsonProperty("numarInregistrare", String)
	public numarInregistrare: string = null;

	@JsonProperty("dataInregistrare", JsonDateConverter)
	public dataInregistrare: Date = null;
	
	@JsonProperty("numeEmitent", String)
	public numeEmitent: string = null;
	
	@JsonProperty("departamentEmitent", String)
	public departamentEmitent: string = null;
	
	@JsonProperty("numarDocumentEmitent", String)
	public numarDocumentEmitent: string = null;
	
	@JsonProperty("dataDocumentEmitent", JsonDateConverter)
	public dataDocumentEmitent: Date = null;
	
	@JsonProperty("tipDocument", String)
	public tipDocument: string = null;
	
	@JsonProperty("codTipDocument", String)
	public codTipDocument: string = null;

	@JsonProperty("trimisPeMail", Boolean)
	public trimisPeMail: boolean = null;

	@JsonProperty("continut", String)
	public continut: string = null;

	@JsonProperty("numarPagini", Number)
	public numarPagini: number = null;

	@JsonProperty("numarAnexe", Number)
	public numarAnexe: number = null;

	@JsonProperty("repartizatCatre", String)
	public repartizatCatre: string = null;

	@JsonProperty("comisieSauGL", String)
	public comisieSauGL: string = null;

	@JsonProperty("proiect", String)
	public proiect: string = null;

	@JsonProperty("necesitaRaspuns", Boolean)
	public necesitaRaspuns: boolean = null;

	@JsonProperty("termenRaspuns", JsonDateConverter)
	public termenRaspuns: Date = null;

	@JsonProperty("numarInregistrareOfRegistruIesiri", String)
	public numarInregistrareOfRegistruIesiri: string = null;

	@JsonProperty("observatii", String)
	public observatii: string = null;

	@JsonProperty("raspunsuriBanciCuPropuneri", String)
	public raspunsuriBanciCuPropuneri: string = null;

	@JsonProperty("nrZileIntrareEmitent", Number)
	public nrZileIntrareEmitent: number = null;

	@JsonProperty("nrZileRaspunsIntrare", Number)
	public nrZileRaspunsIntrare: number = null;

	@JsonProperty("nrZileRaspunsEmitent", Number)
	public nrZileRaspunsEmitent: number = null;

	@JsonProperty("nrZileTermenDataRaspuns", Number)
	public nrZileTermenDataRaspuns: number = null;

	@JsonProperty("anulat", Boolean)
	public anulat: boolean = null;

	@JsonProperty("motivAnulare", String)
	public motivAnulare: string = null;

	@JsonProperty("inchis", Boolean)
	public inchis: boolean = null;

	@JsonProperty("trebuieAnulat", Boolean, true)
	public trebuieAnulat: boolean = null;

	@JsonProperty("subactivityName", String)
	public subactivityName: string = null;

	public get dataInregistrareForDisplay(): string {
		return DateUtils.formatForDisplay(this.dataInregistrare);
	}

	public get dataDocumentEmitentForDisplay(): string {
		return DateUtils.formatForDisplay(this.dataDocumentEmitent);
	}

	public get trimisPeMailForDisplay(): string {
		if (this.trimisPeMail) {
			return "LABELS.YES";
		}
		return "LABELS.NO";
	}

	public get necesitaRaspunsForDisplay(): string {
		if (this.necesitaRaspuns) {
			return "LABELS.YES";
		}
		return "LABELS.NO";
	}

	public get termenRaspunsForDisplay(): string {
		return DateUtils.formatForDisplay(this.termenRaspuns);
	}

	public get raspunsuriBanciCuPropuneriForDisplay(): string {
		if (this.raspunsuriBanciCuPropuneri === RaspunsuriBanciCuPropuneriEnum.DA) {
			return "LABELS.YES";
		} else if (this.raspunsuriBanciCuPropuneri === RaspunsuriBanciCuPropuneriEnum.NU) {
			return "LABELS.NO";
		} else {
			return "LABELS.NA";
		}
	}

	public get anulatForDisplay(): string {
		if (this.anulat) {
			return "LABELS.YES";
		}
		return "LABELS.NO";
	}

	public get inchisForDisplay(): string {
		if (this.inchis) {
			return "LABELS.YES";
		}
		return "LABELS.NO";
	}

	public get luna(): number {
		return this.dataInregistrare.getMonth();
	}

	public get lunaForDisplay(): string {
		return "LABELS." + DateConstants.getMonthCodeByIndex(this.dataInregistrare.getMonth());
	}

	public get numarInregistrareForDisplay(): string {
		let numarInregistrarePerAn: string = this.numarInregistrare;
		let numarInregistrareSplitArray: string[] = numarInregistrarePerAn.split("~");

		return numarInregistrareSplitArray[0];
	}

	public get numarInregistrareForFilter(): number {
		let numarInregistrarePerAn: string = this.numarInregistrare;
		let numarInregistrareSplitArray: string[] = numarInregistrarePerAn.split("~");

		return Number(numarInregistrareSplitArray[0]);
	}
	
	public get numarInregistrareOfRegistruIesiriForFilter(): number {
		if (this.numarInregistrareOfRegistruIesiri != null) {
			let numarInregistrarePerAn: string = this.numarInregistrareOfRegistruIesiri;
			let numarInregistrareSplitArray: string[] = numarInregistrarePerAn.split("~");

			return Number(numarInregistrareSplitArray[0]);
		} else {
			return -1;
		} 
	}
}

File: CloudFlow/src/ro/cloudSoft/cloudDoc/presentation/client/shared/model/registruIntrariIesiri/RegistruIntrariViewModel.java
package ro.cloudSoft.cloudDoc.presentation.client.shared.model.registruIntrariIesiri;

import java.util.Date;
import java.util.List;

import ro.cloudSoft.cloudDoc.domain.AtasamentModel;
import ro.cloudSoft.cloudDoc.presentation.client.shared.model.registruIntrariIesiri.RegistruIntrariModel.RaspunsuriBanciCuPropuneriEnum;

public class RegistruIntrariViewModel {

	private Long id;
	private String numarInregistrare;
	private Date dataInregistrare;
	private String numeEmitent;
	private String departamentEmitent;
	private String numarDocumentEmitent;
	private Date dataDocumentEmitent;
	private String tipDocument;
	private String codTipDocument;
	private boolean trimisPeMail;
	private String continut;
	private Integer numarPagini;
	private Integer numarAnexe;
	private String repartizatCatre;
	private String comisieSauGL;
	private String proiect;
	private boolean necesitaRaspuns;
	private Date termenRaspuns;
	private String numarInregistrareOfRegistruIesiri;
	private String observatii;
	private RaspunsuriBanciCuPropuneriEnum raspunsuriBanciCuPropuneri;
	private Integer nrZileIntrareEmitent;
	private Integer nrZileRaspunsIntrare;
	private Integer nrZileRaspunsEmitent;
	private Integer nrZileTermenDataRaspuns;
	private boolean anulat;
	private String motivAnulare;
	private boolean inchis;
	private boolean trebuieAnulat;
	private List<AtasamentModel> atasamente;
	private String subactivityName;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getNumarInregistrare() {
		return numarInregistrare;
	}

	public void setNumarInregistrare(String numarInregistrare) {
		this.numarInregistrare = numarInregistrare;
	}

	public Date getDataInregistrare() {
		return dataInregistrare;
	}

	public void setDataInregistrare(Date dataInregistrare) {
		this.dataInregistrare = dataInregistrare;
	}

	public String getNumeEmitent() {
		return numeEmitent;
	}

	public void setNumeEmitent(String numeEmitent) {
		this.numeEmitent = numeEmitent;
	}

	public String getDepartamentEmitent() {
		return departamentEmitent;
	}

	public void setDepartamentEmitent(String departamentEmitent) {
		this.departamentEmitent = departamentEmitent;
	}

	public String getNumarDocumentEmitent() {
		return numarDocumentEmitent;
	}

	public void setNumarDocumentEmitent(String numarDocumentEmitent) {
		this.numarDocumentEmitent = numarDocumentEmitent;
	}

	public Date getDataDocumentEmitent() {
		return dataDocumentEmitent;
	}

	public void setDataDocumentEmitent(Date dataDocumentEmitent) {
		this.dataDocumentEmitent = dataDocumentEmitent;
	}

	public String getTipDocument() {
		return tipDocument;
	}

	public void setTipDocument(String tipDocument) {
		this.tipDocument = tipDocument;
	}

	public String getCodTipDocument() {
		return codTipDocument;
	}

	public void setCodTipDocument(String codTipDocument) {
		this.codTipDocument = codTipDocument;
	}

	public boolean isTrimisPeMail() {
		return trimisPeMail;
	}

	public void setTrimisPeMail(boolean trimisPeMail) {
		this.trimisPeMail = trimisPeMail;
	}

	public String getContinut() {
		return continut;
	}

	public void setContinut(String continut) {
		this.continut = continut;
	}

	public Integer getNumarPagini() {
		return numarPagini;
	}

	public void setNumarPagini(Integer numarPagini) {
		this.numarPagini = numarPagini;
	}

	public Integer getNumarAnexe() {
		return numarAnexe;
	}

	public void setNumarAnexe(Integer numarAnexe) {
		this.numarAnexe = numarAnexe;
	}

	public String getRepartizatCatre() {
		return repartizatCatre;
	}

	public void setRepartizatCatre(String repartizatCatre) {
		this.repartizatCatre = repartizatCatre;
	}

	public String getComisieSauGL() {
		return comisieSauGL;
	}

	public void setComisieSauGL(String comisieSauGL) {
		this.comisieSauGL = comisieSauGL;
	}

	public String getProiect() {
		return proiect;
	}

	public void setProiect(String proiect) {
		this.proiect = proiect;
	}

	public boolean isNecesitaRaspuns() {
		return necesitaRaspuns;
	}

	public void setNecesitaRaspuns(boolean necesitaRaspuns) {
		this.necesitaRaspuns = necesitaRaspuns;
	}

	public Date getTermenRaspuns() {
		return termenRaspuns;
	}

	public void setTermenRaspuns(Date termenRaspuns) {
		this.termenRaspuns = termenRaspuns;
	}

	public String getNumarInregistrareOfRegistruIesiri() {
		return numarInregistrareOfRegistruIesiri;
	}

	public void setNumarInregistrareOfRegistruIesiri(String numarInregistrareOfRegistruIesiri) {
		this.numarInregistrareOfRegistruIesiri = numarInregistrareOfRegistruIesiri;
	}

	public String getObservatii() {
		return observatii;
	}

	public void setObservatii(String observatii) {
		this.observatii = observatii;
	}

	public RaspunsuriBanciCuPropuneriEnum getRaspunsuriBanciCuPropuneri() {
		return raspunsuriBanciCuPropuneri;
	}

	public void setRaspunsuriBanciCuPropuneri(RaspunsuriBanciCuPropuneriEnum raspunsuriBanciCuPropuneri) {
		this.raspunsuriBanciCuPropuneri = raspunsuriBanciCuPropuneri;
	}

	public Integer getNrZileIntrareEmitent() {
		return nrZileIntrareEmitent;
	}

	public void setNrZileIntrareEmitent(Integer nrZileIntrareEmitent) {
		this.nrZileIntrareEmitent = nrZileIntrareEmitent;
	}

	public Integer getNrZileRaspunsIntrare() {
		return nrZileRaspunsIntrare;
	}

	public void setNrZileRaspunsIntrare(Integer nrZileRaspunsIntrare) {
		this.nrZileRaspunsIntrare = nrZileRaspunsIntrare;
	}

	public Integer getNrZileRaspunsEmitent() {
		return nrZileRaspunsEmitent;
	}

	public void setNrZileRaspunsEmitent(Integer nrZileRaspunsEmitent) {
		this.nrZileRaspunsEmitent = nrZileRaspunsEmitent;
	}

	public Integer getNrZileTermenDataRaspuns() {
		return nrZileTermenDataRaspuns;
	}

	public void setNrZileTermenDataRaspuns(Integer nrZileTermenDataRaspuns) {
		this.nrZileTermenDataRaspuns = nrZileTermenDataRaspuns;
	}

	public boolean isAnulat() {
		return anulat;
	}

	public void setAnulat(boolean anulat) {
		this.anulat = anulat;
	}

	public String getMotivAnulare() {
		return motivAnulare;
	}

	public void setMotivAnulare(String motivAnulare) {
		this.motivAnulare = motivAnulare;
	}

	public boolean isInchis() {
		return inchis;
	}

	public void setInchis(boolean inchis) {
		this.inchis = inchis;
	}
	
	public List<AtasamentModel> getAtasamente() {
		return atasamente;
	}

	public void setAtasamente(List<AtasamentModel> atasamente) {
		this.atasamente = atasamente;
	}

	public boolean isTrebuieAnulat() {
		return trebuieAnulat;
	}

	public void setTrebuieAnulat(boolean trebuieAnulat) {
		this.trebuieAnulat = trebuieAnulat;
	}

	public String getSubactivityName() {
		return subactivityName;
	}

	public void setSubactivityName(String subactivityName) {
		this.subactivityName = subactivityName;
	}

}
